File: packages/core-go/address/detect.go
package address

import (
	"github.com/stellar/go/strkey"
)

func Detect(address string) string {
	if strkey.IsValidEd25519PublicKey(address) {
		return "G"
	}
	if strkey.IsValidMuxedAccountEd25519PublicKey(address) {
		return "M"
	}
	if _, err := strkey.Decode(strkey.VersionByteContract, address); err == nil {
		return "C"
	}
	return "invalid"
}

File: packages/core-dart/lib/src/address/detect.dart
import 'codes.dart';

AddressKind detect(String address) {
  if (address.startsWith('G') && address.length == 56) return AddressKind.g;
  if (address.startsWith('M') && address.length == 95) return AddressKind.m;
  if (address.startsWith('C') && address.length == 56) return AddressKind.c;
  return AddressKind.invalid;
}
